### PR TITLE
feat(admin-portal): Module switcher

### PR DESCRIPTION
**Pull request by saevarma**
# Admin portal module switcher

[Notion](https://www.notion.so/Implement-dropdown-switcher-d9f3d35356bb4f03a822ba99a885e920)

## What

Adding module switcher navigation both desktop and mobile vesions.

## Why

To navigate between modules.

## Screenshots / Gifs

image

image

image

image

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/portals/admin/src/components/ModuleSwitcher/ModuleSwitcherItems.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
+import React from 'react'
+import { Link } from 'react-router-dom'
+
+import { Box, Icon, IconMapIcon, Stack, Text } from '@island.is/island-ui/core'
+import { useLocale } from '@island.is/localization'
+import { m, useActiveModule, useNavigation } from '@island.is/portals/core'
+
+interface ModuleItemsProps {
+  mobile?: boolean
+  onNavigation: () => void
```
Comment (snaerseljan) [inline apps/portals/admin/src/components/ModuleSwitcher/ModuleSwitcherItems.tsx:10]: nit: can do short hand `onNavigation(): void`

File: apps/portals/admin/src/components/ModuleSwitcher/ModuleSwitcherMobile.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            <Text>
+              {formatMessage(
+                activeModule ? activeModule.name : masterNavigation.name,
+              )}
+            </Text>
+          </Box>
+          <Box display="flex" alignItems="center" marginLeft={2}>
+            <Button
+              colorScheme="negative"
+              circle={true}
```
Comment (snaerseljan) [inline apps/portals/admin/src/components/ModuleSwitcher/ModuleSwitcherMobile.tsx:49]: nit: shorthand true prop

File: apps/portals/admin/src/components/ModuleSwitcher/ModuleSwitcherMobile.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+              size="small"
+              icon="chevronDown"
+              title={formatMessage(m.openModuleSwitcher)}
+            ></Button>
```
Comment (snaerseljan) [inline apps/portals/admin/src/components/ModuleSwitcher/ModuleSwitcherMobile.tsx:53]: nit: shorthand close element

File: libs/portals/admin/core/src/lib/messages.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import { defineMessages } from 'react-intl'
+
+export const m = defineMessages({
+  dashboard: {
+    id: 'portals:dashboard',
```
Comment (snaerseljan) [inline libs/portals/admin/core/src/lib/messages.ts:5]: In the dashboard PR I set the namespace to `admin-portal:key` since it is admin core module

**Comment by snaerseljan** [comment]
According to design the arrow button should be white. Ignore if intentional

Figma
image

current branch
image

**Comment by snaerseljan** [comment]
Talked to the design team regarding active and hover state because I noticed and underline on the hover item.
This is now the updated design with no underline on link hover.
image

**Review by snaerseljan — APPROVED** [review]
Great work. Had a few nits and some observations. Me and @eirikurn discussed if we should have the hole switcher as clickable since its borders indicate that this is a button.

But aside from that awesome job 🥇

File: libs/portals/admin/core/src/lib/messages.ts (hunk 4 above)
Comment (saevarma) [reply libs/portals/admin/core/src/lib/messages.ts:5]: Absolutlety, fixed.

File: apps/portals/admin/src/components/ModuleSwitcher/ModuleSwitcherItems.tsx (hunk 1 above)
Comment (saevarma) [reply apps/portals/admin/src/components/ModuleSwitcher/ModuleSwitcherItems.tsx:10]: Nice!

File: apps/portals/admin/src/components/ModuleSwitcher/ModuleSwitcherDesktop.css.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+export const menuButton = style({
+  width: '100%',
+})
+
+export const backdrop = style({
+  background: 'rgba(242, 247, 255, 0.7)',
+  zIndex: theme.zIndex.belowModal,
+  position: 'fixed',
+  top: 0,
```
[the remark this answers is not in the document]
Comment (saevarma) [reply apps/portals/admin/src/components/ModuleSwitcher/ModuleSwitcherDesktop.css.ts:23]: Forgot to update after I learnt about inset, thanks!

File: apps/portals/admin/src/components/ModuleSwitcher/ModuleSwitcherDesktop.css.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import { style } from '@vanilla-extract/css'
+
+import { theme } from '@island.is/island-ui/theme'
+
+export const container = style({
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'spaceBetween',
+  width: '290px',
```
[the remark this answers is not in the document]
Comment (saevarma) [reply apps/portals/admin/src/components/ModuleSwitcher/ModuleSwitcherDesktop.css.ts:9]: Yes this was tricky on the desktop as it is not following the grid system, and we don't want it to change size based on how long the module name is. So I ended up with using fixed width on this, but I totally agree I don't like settings hardcoded widths.

**Comment by saevarma** [comment]
> According to design the arrow button should be white. Ignore if intentional
> 
> Figma image
> 
> current branch image

Yes we should absolutely do that.

**Pull request merged**